keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,659 @@
+  - Components depending on the feature gate:
+      - kube-apiserver
+      - kubelet
+
+###### Does enabling the feature change any default behavior?
+
+No - AppArmor has been enabled by default since Kubernetes v1.4.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Containers already running with AppArmor enforcement will continue to do so, but on restart
+will fallback to the container runtime default. Pods created with AppArmor disabled will have their
+fields & annotations stripped.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Newly started or restarted containers in pods that still have the AppArmor field/annotations will
+have the specified AppArmor profile applied, rather than the runtime default.
+
+###### Are there any tests for feature enablement/disablement?
+
+No.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The [Version Skew Strategy](#version-skew-strategy) section covers this point.
+Running workloads should have no impact as the Kubelet will support either the
+existing annotations or the new fields introduced by this KEP.
+
+Disabling the AppArmor feature will cause the container runtimes to apply the runtime default
+profile (except for privileged pods). In cases where a user was expecting to apply a custom profile
+(or explicitly unconfined), this could break the workload.
+
+###### What specific metrics should inform a rollback?
+
+An increase in pod validation errors can indicate issues with the field translation. These would
+show up as `code=400` (Bad Request) errors in `apiserver_request_total`.
+
+The following errors could indicate problems with how kubelets are interpreting AppArmor profiles.
+* `started_containers_errors_total`
+* `started_pods_errors_total`
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Automated tests will cover the scenarios with and without the changes proposed
+on this KEP. As defined under [Version Skew Strategy](#version-skew-strategy),
+we are assuming the cluster may have kubelets with older versions (without
+this KEP' changes), therefore this will be covered as part of the new tests.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+The promotion of AppArmor to GA would deprecate the beta annotations as described in the
+[Version Skew Strategy](#version-skew-strategy).
+
+### Monitoring requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The feature is built into the kubelet and api server components. No metric is
+planned at this moment. The way to determine usage is by checking whether the
+pods/containers have a AppArmorProfile set.
+
+###### How can someone using this feature know that it is working for their instance?
```

Comment:
I guess maybe also not apparmor itself but the field <> annotation bit?

Reply:
Sorry, I don't quite understand this. Do you mean how can they be sure that the pod had the correct API bits set? As in, get the pod yaml and confirm the annotation & fields match?